source/boundary_composition/interface.cc

```diff
@@ -47,6 +49,32 @@ namespace aspect
     {}
 
     template <int dim>
+    double
+    Interface<dim>::composition (const GeometryModel::Interface<dim> &/*geometry_model*/,
+                                 const types::boundary_id             boundary_indicator,
+                                 const Point<dim>                    &position,
+                                 const unsigned int                   compositional_field) const
+    {
+      /**
+       * Call the new-style function without the geometry model
+       * to maintain backwards compatibility. Normally the derived class should
+       * override the called function, but if it overrides this function, the
+       * new interface will not be used.
+       */
+
+      return boundary_composition(boundary_indicator,position,compositional_field);
+    }
+
+    template <int dim>
+    double
+    Interface<dim>::boundary_composition (const types::boundary_id             /*boundary_indicator*/,
+                                          const Point<dim>                    &/*position*/,
+                                          const unsigned int                   /*compositional_field*/) const
+    {
+      return std::numeric_limits<double>::quiet_NaN();
```

Comment:
You are right, nothing good can happen if this function is used to calculate something. But for reasons I will discuss below I need a way to check if it is implemented in a particular plugin, without aborting the computation in case it is not. That is why I decided for a default implementation that returns something specifically useless that I can test.